Split ajasalminen.ssl_certificate for local mkcert

humble-ahitofel opened the pull request:
- ssl_certificate role is split into production  and local tasks.
      the local tasks is a new addition that uses mkcert tool to generate
      "valid" certificates for local development enviroment.
      For more information:
      https://github.com/FiloSottile/mkcert

humble-ahitofel committed:
Split ajasalminen.ssl_certificate for local mkcert

    - ssl_certificate role is split into production  and local tasks.
      the local tasks is a new addition that uses mkcert tool to generate
      "valid" certificates for local development enviroment.
      For more information:
      https://github.com/FiloSottile/mkcert

humble-ahitofel committed:
Two fixes to mkcert path

- Apache/Nginx ssl folders don't necessarly exist when copying the
  folder, now folder's get created before copying the certficiates.
- webserver in notification is not ansible built-in callback, removed it.

ajsalminen committed:
Delete .galaxy_install_info

Install info should be outside version control.

ajsalminen commented on tasks/local.yml:
```diff
@@ -0,0 +1,105 @@
+  get_url:
+    url: '{{ ssl_certificate_mkcert_url }}'
+    dest: '{{ ssl_certificate_mkcert_target_dir }}/mkcert'
+    mode: 0750
```
Using checksum would be good so we can be sure the file is the correct one.

humble-ahitofel replied:
Added self-generated checksum to get_url task.

ajsalminen commented on tasks/local.yml:
```diff
@@ -0,0 +1,105 @@
+    -cert-file {{ ssl_certificate_mkcert_generated_cert_dir }}/{{ ssl_certificate_chained }}
+    -key-file {{ ssl_certificate_mkcert_generated_cert_dir }}/{{ ssl_certificate_key }}
+    {{ ssl_certificate_key_common_name }}.local
+    *.{{ ssl_certificate_key_common_name }}.local 
```
Making the certificate domains configurable would be nice so that it'll be possible to use any TLD or suffix.

humble-ahitofel replied:
I added optional 'ssl_certificate_mkcert_additional_names' variable that allows you to specify additonal domains.

ajsalminen commented on tasks/local.yml:
```diff
@@ -0,0 +1,105 @@
+---
+
+- name: Create local directory for mkcert stuff
```
Current role isn't entirely consistent about it but let's end task names as full sentences with a period since that's the style for most of the tasks right now.

humble-ahitofel replied:
I made some senteces a bit more meaningful and added periods to all task names.

ajsalminen commented on tasks/local.yml:
```diff
@@ -0,0 +1,105 @@
+
+# This might be run before nginx or apache2 install's, so correct folders need
+# to be created.
+# create a directory if it doesn't exist
```
Seems redundant with the task name. :)

humble-ahitofel replied:
Removed that and also some other comment (can't remember what).

ajsalminen commented on tasks/local.yml:
```diff
@@ -0,0 +1,105 @@
+  get_url:
+    url: '{{ ssl_certificate_mkcert_url }}'
+    dest: '{{ ssl_certificate_mkcert_local_dir }}/mkcert'
+    mode: 0750
```
Could modes use the symbolic notation everywhere.

humble-ahitofel replied:
Yeah, those are much more readable. Rewrote every task to use symbolic notation.

humble-ahitofel committed:
Added option to specify additional domains

- Added checksum check for the mkcert download
- Domain suffix can be now overwritten
- Removed some  unecessary comments.
- File mode definitions are now symbolic
- Still needs testing, this not the *working* commit for these additions

humble-ahitofel committed:
Added periods to all task sentences.

humble-ahitofel committed:
Fixed combining multiple domains.

- Also added periods to all task sentences.
